### DISCUSSION TITLE
Radiomaster TX12 Problem

### DISCUSSION THREAD

**Discussion by JPMat78** [pascallanger/DIY-Multiprotocol-TX-Module#635] [Q&A]
Hi, I just read the discution "Radiomaster TX12: Status no telemetry #613". I have the same problem. I received a new TX12 from Banggood. I could not bind so I have updated successfully the internal module with "mm-stm-cc2500-aetr-v1.3.3.0.bin". But from this time I have "No Telemetry" and impossible to flash again the internal module, I tried with "mm-stm-cc2500-64-aetr-v1.3.3.0.bin" 
 and also with the factory one, on Radiomaster website but every time I have "no sync" message. If somebody have a solution please let me know. I live in France, do you know if any shop can fix this radio ??

**Comment by JPMat78** [comment]
Actually I send back the radio to Banggood. It is too much waist of time to fix a radio with such default.
For information they send me from Czech republic and I have to send back in China.

**Reply by krisdover** [reply]
FWIW, I had the same issue after flashing the `mm-stm-cc2500-64-aetr-v1.3.3.0.bin` release I got here. I've since checked out the project and built my own v1.3.3.0 binary which seems to work fine.

**Reply by benlye** [reply]
This doesn't really make any sense to me.  I took a look at our build scripts and confirmed that the `-64-` FW builds really do use the 64KB 'C8' board version.  

I don't have a 64KB TX12 but I do have a 64KB external module and I've confirmed that the `mm-stm-cc2500-64-aetr-v1.3.3.0.bin` from downloads.multi-module.org works just fine on that (and any 128KB build fails, as expected).

**Reply by krisdover** [reply]
@benlye I've re-flashed the `mm-stm-cc2500-64-aetr-v1.3.3.0.bin` binary from the release to double check and sadly I was unable to reproduce the "Status no telemetry" error message I was getting before in model setup, so I'm not sure what has changed.

Roughly what I did earlier when I encountered the issue was:

* updated the TX12 to EdgeTX
* went to setup a model and saw the "Advise upgrade" status message (sorry I can't remember the OEM firmware version).
* initially downloaded and flashed the 128k binary which didn't work.
* searched for the "Status no telemetry" error and found issue #613 which lead me to strip down my radio to discover I had the 64k STM chip.
* downloaded and flashed the 64k binary which still didn't work (Status no telemetry).
* noticed the similarity in binary file sizes and ran a SHA1 hash of the 64k (`25c5733355806d35cc6b8f3509d364b619f702cb`) and 128k (`f8287bf83c4dcae4d11001ebdd78a939e41ac429`) files to make sure they were indeed different.
* downloaded and flashed the older `mm-cc2500-64-v1.3.2.89.bin` provided on issue #613 which worked.
* checked out the source for the project, built and flashed v1.3.3.0 which worked.

I don't know if my initial 128k install left the module is a bad state but I'm out of ideas. Thanks, and apologies for the wild goose chase.

**Comment by sas58567** [comment]
I'm a new with flashing. I use TX12 64k. I update otx and cc2500 from factory to otx 2.3.15 and cc2500 1.3.3.20. But if I try to select protocol multi/mjxq then I got "prot. invalid" and no sub protocol selection. Also I tryed return to cc2500 1.3.2.6 from radiomaster site and then RF is not working completely - I tryed to use sys/tools/spectrum and got no spectre. Then I back to cc2500 1.3.3.20 and got spectre back. I remember that MJXq worked from factory.
Why MJXq is not working in otx 2.3.15 and cc2500 1.3.3.20?

**Reply by pascallanger** [reply]
I would say that the issue is in OTX 2.3.15 . Try to flash an older version of OTX instead of playing with the module versions.

**Reply by sas58567** [reply]
Thank you. I downgraded OTX to 2.3.11 and for the 1st view nothing changed. I had troubles with flashing and drivers for STM bootloader device, but the Zadig tool helped me to reinstall the driver. Then OTX Companion found the divice and flashed it correctly. So I still didn't saw the E010 in MJXq - I saw only blank field without selection in sub-protocol menu. But I tryed to enter into blank field by pressing the rotor-button on TX12 and tryed to switch sub-protocol and bingo! - E010 appears after several movements of the rotor-button. It still not worked with my E010 red board drone. Then I tryed to upgrade OTX back to 2.3.15 and it work with selection of the sub-protocol E010 and blank values in same manner as I write above.
  Also I found that in my TX12 64k cc2500 has no E010R5 protocol. When I update cc2500 to non-64k version from downloads-multi-module site, then I have E010R5 in protocol list. Is E010R5 protocol excluded from 64k version? I can't use non-64k version because when I update my TX12 to it, then seems like RF is not working - no spectre in spectre lua tool.

**Reply by pascallanger** [reply]
Your CC2500 only module can't run a 128KB firmware.
The E010R5 protocol uses the cyrf6936 for the emulation which you don't have in your module.
 Documentation : https://github.com/pascallanger/DIY-Multiprotocol-TX-Module/blob/master/Protocols_Details.md#e010r5---81

**Comment by Aleksandr1975** [comment]
Добрый день может кто нибудь помочь купил новый radiomaster tx12 EdgeTX-ELRS хотел привязать приёмник FS-AI6b ничего не получилось в статусе пишет нет телеметрии мульти модуль аппа не видит при порошивки выдаёт ошибка FM update error Device not responding

**Comment by JPMat78** [comment]
Hi, if your TX12 is the ELRS version, you can use only ELRS receivers. You
do not have a multi module. But on the external bay you can buy and put an
external multi module. Good luck !

Le sam. 11 mars 2023 à 11:05, Aleksandr1975 ***@***.***> a
écrit :

**Reply by Aleksandr1975** [reply]
Большое спасибо за помощь пусть ваш день будет удачным
